File: Modules/AdvantShop.Module.Roistat/Content/scripts/admin/roistatSettings.js
; (function (ng) {
    'use strict';

    var roistatSettingsCtrl = function ($http, toaster) {
        var ctrl = this;
        
        ctrl.$onInit = function () {
            ctrl.getSettings();
        };

        ctrl.getSettings = function () {
            $http.get('../roistatSettings/getSettings').then(function (response) {
                ctrl.settings = response.data;
                ctrl.haveAccount =
                    (ctrl.settings.RoistatLogin != null ||
                        ctrl.settings.RoistatPassword != null ||
                        ctrl.settings.RoistatScript != null)
                    ? 'true'
                    : 'false';
            });
        }

        ctrl.register = function() {
            ctrl.haveAccount = 'true';
            ctrl.saveSettings();
        }

        ctrl.saveSettings = function () {
            $http.post('../roistatSettings/saveSettings', ctrl.settings).then(function (response) {
                if (response.data.result === true) {
                   toaster.pop('success', '', 'Настройки сохранены');
               } else {
                   toaster.pop('error', '', 'Ошибка при сохранении настроек');
               }
            });
        }
    };

    roistatSettingsCtrl.$inject = ['$http', 'toaster'];

    ng.module('roistatSettings', [])
        .controller('roistatSettingsCtrl', roistatSettingsCtrl)
        .component('roistatSettings', {
            templateUrl: '../modules/roistat/content/scripts/admin/roistatSettings.html',
            controller: 'roistatSettingsCtrl'
        });


})(window.angular);

File: Modules/AdvantShop.Module.Roistat/Domain/RoistatSettings.cs
using AdvantShop.Core.Modules;

namespace AdvantShop.Module.Roistat.Domain
{
    public class RoistatSettings
    {
        public static string RoistatScript
        {
            get { return ModuleSettingsProvider.GetSettingValue<string>("RoistatScript", Roistat.ModuleId); }
            set { ModuleSettingsProvider.SetSettingValue("RoistatScript", value, Roistat.ModuleId); }
        }

        /// <summary>
        /// Имя пользователя (Указывается в настройках интеграции)
        /// </summary>
        public static string RoistatLogin
        {
            get { return ModuleSettingsProvider.GetSettingValue<string>("RoistatLogin", Roistat.ModuleId); }
            set { ModuleSettingsProvider.SetSettingValue("RoistatLogin", value, Roistat.ModuleId); }
        }

        /// <summary>
        /// Пароль пользователя (Указывается в настройках интеграции)
        /// </summary>
        public static string RoistatPassword
        {
            get { return ModuleSettingsProvider.GetSettingValue<string>("RoistatPassword", Roistat.ModuleId); }
            set { ModuleSettingsProvider.SetSettingValue("RoistatPassword", value, Roistat.ModuleId); }
        }
    }
}
